packages/core/agent/src/forever/forever-daemon.ts

```diff
@@ -57,6 +56,10 @@ export class ForeverDaemon implements Daemon {
 
   async start(profile: string): Promise<ProcessInfo> {
     if (!(await this.isRunning(profile))) {
+      if (isLocked({ lockKey: profile, root: path.join(this._rootDir, 'profile', profile) })) {
```

Comment:
@dmaretskyi see `service-host.ts`

Reply:
how about `serviceHost.isLockTaken()` to encapsulate the lock